src/main/java/seedu/sudohr/model/employee/UniqueEmployeeList.java

```diff
@@ -1,134 +1,281 @@
 
 /**
  * A list of employees that enforces uniqueness between its elements and does not allow nulls.
- * An employee is considered unique by comparing using {@code Employee#isSameEmployee(Employee)}.
- * As such, adding and updating of employees uses Employee#isSameEmployee(Employee) for equality
- * so as to ensure that the employee being added or updated is
- * unique in terms of identity in the UniqueEmployeeList. However, the removal of an employee
- * uses Employee#equals(Object) so as to ensure that the employee with exactly the same fields will be removed.
+ * An employee is considered unique by comparing using {@code Employee#isSameEmployee(Employee)}. As such, adding and
+ * updating of employees uses Employee#isSameEmployee(Employee) for equality to ensure that the person being added or
+ * updated isunique in terms of identity in the UniqueEmployeeList. However, the removal of a person uses
+ * Employee#equals(Object) to ensure that the person with exactly the same fields will be removed.
  *
  * Supports a minimal set of list operations.
  *
  * @see Employee#isSameEmployee(Employee)
  */
 public class UniqueEmployeeList implements Iterable<Employee> {
-
     private final ObservableList<Employee> internalList = FXCollections.observableArrayList();
     private final ObservableList<Employee> internalUnmodifiableList =
             FXCollections.unmodifiableObservableList(internalList);
-
     /**
-     * Returns true if the list contains an equivalent employee as the given argument.
+     * Returns true if the list contains an equivalent person as the given argument.
      */
     public boolean contains(Employee toCheck) {
         requireNonNull(toCheck);
         return internalList.stream().anyMatch(toCheck::isSameEmployee);
     }
 
     /**
-     * Adds a employee to the list.
-     * The employee must not already exist in the list.
+     * Returns true if the list contains an equivalent person as the given argument,
+     * excluding the specified person.
+     */
+    public boolean contains(Employee toCheck, Employee toExclude) {
+        requireNonNull(toCheck);
+        requireNonNull(toExclude);
+        for (Employee employee : internalList) {
+            if (!employee.isSameEmployee(toExclude)) {
+                if (employee.isSameEmployee(toCheck)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if the list contains a different person with the same email as the given argument.
+     */
+    public boolean sharesEmail(Employee toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::emailClashes);
+    }
+
+    /**
+     * Returns true if the list contains a different person with the same email as the given argument,
+     * excluding the specified person.
+     */
+    public boolean sharesEmail(Employee toCheck, Employee toExclude) {
```

Comment:
like `internalist.stream().filter(toCheck::!isSameEmployee).anyMatch(...)`

Reply:
Discussed offline. This way of checking may result in unwanted behavior. E.g editing a person to have different id. This check will flag out there's a duplicated email / phone number because it cannot 'exclude itself' from the checks since id has changed (recall we now define equality between employees strictly by id)

Reply:
Edit: Thought reviewer was referring to method above.

Thanks, you are right, we can improve to the definition u propose for consistency.